File: maven-root/mtsa/src/main/java/RobotProtocol/src/main/java/robotProtocol/messages/DestinationReachedMessage.cpp
/*
 * DestinationReachedMessage.cpp
 *
 *  Created on: Apr 16, 2013
 *      Author: mariano
 */

#include "DestinationReachedMessage.h"
#include <stdlib.h>
#include <stdio.h>

namespace RobotProtocol {

DestinationReachedMessage DestinationReachedMessage::staticMessage = DestinationReachedMessage(0,0);

Message *DestinationReachedMessage::deserialize(char *serializedMessage){
	DestinationReachedMessage::staticMessage.initialize(serializedMessage);
	return &(DestinationReachedMessage::staticMessage);
}

DestinationReachedMessage::DestinationReachedMessage(int to, int from)
:Message(MESSAGE_ID, 1, to, from){}

DestinationReachedMessage::DestinationReachedMessage(char *serializedMessage){
	initialize(serializedMessage);
}

DestinationReachedMessage::~DestinationReachedMessage() {
}

int DestinationReachedMessage::getMessageLength(){
	return Message::getMessageLength();
}

void DestinationReachedMessage::readData(char **serializedMessageIndex){}

void DestinationReachedMessage::writeData(char **serializedMessageIndex){}

} /* namespace RobotProtocol */

File: maven-root/mtsa/src/main/java/MTSAEnactment/ar/uba/dc/lafhis/enactment/robot/N6/robotProtocol/DestinationReachedMessage.java
package MTSAEnactment.ar.uba.dc.lafhis.enactment.robot.N6.robotProtocol;

public class DestinationReachedMessage extends Message {	
	public static int MESSAGE_ID				= 203;
	
	public DestinationReachedMessage(int to, int from){
		super(MESSAGE_ID, 1, to, from);
	}
	
	public DestinationReachedMessage(String serializedMessage){
		super(serializedMessage);
	}
	
	@Override
	public int getMessageLength() {
		return super.getMessageLength();
	}
	
	
	@Override
	protected void readData(String inputString){
	}

	@Override
	protected String writeData(){
		return "";
	}

	@Override
	protected boolean hasReply() {
		return false;
	}
	
}
